src/main/java/com/autotune/analyzer/workerimpl/BulkJobManager.java

```diff
@@ -0,0 +1,229 @@
+import com.autotune.common.datasource.DataSourceManager;
+import com.autotune.common.k8sObjects.TrialSettings;
+import com.autotune.common.utils.CommonUtils;
+import com.autotune.database.service.ExperimentDBService;
+import com.autotune.operator.KruizeDeploymentInfo;
+import com.autotune.utils.KruizeConstants;
+import com.autotune.utils.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.ProtocolException;
+import java.net.URL;
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+public class BulkJobManager implements Runnable {
+    private static final Logger LOGGER = LoggerFactory.getLogger(BulkJobManager.class);
+
+    private String jobID;
+    private Map<String, BulkJobStatus> jobStatusMap;
+    private BulkInput bulkInput;
+
+    public BulkJobManager(String jobID, Map<String, BulkJobStatus> jobStatusMap, BulkInput payload) {
+        this.jobID = jobID;
+        this.jobStatusMap = jobStatusMap;
+        this.bulkInput = payload;
+    }
+
+    public static List<String> appendExperiments(List<String> allExperiments, String experimentName) {
+        allExperiments.add(experimentName);
+        return allExperiments;
+    }
+
+    @Override
+    public void run() {
+        try {
+
+            String uniqueKey = null;
+            // Process labels in the 'include' section
+            if (this.bulkInput.getFilter() != null && this.bulkInput.getFilter().getInclude() != null) {
+                // Initialize StringBuilder for uniqueKey
+                StringBuilder includeLabelsBuilder = new StringBuilder();
+                Map<String, String> includeLabels = this.bulkInput.getFilter().getInclude().getLabels();
+                if (includeLabels != null && !includeLabels.isEmpty()) {
+                    includeLabels.forEach((key, value) ->
+                            includeLabelsBuilder.append(key).append("=").append("\"" + value + "\"").append(",")
+                    );
+                    // Remove trailing comma
+                    if (includeLabelsBuilder.length() > 0) {
+                        includeLabelsBuilder.setLength(includeLabelsBuilder.length() - 1);
+                    }
+                    LOGGER.info("Include Labels: " + includeLabelsBuilder.toString());
+                    uniqueKey = includeLabelsBuilder.toString();
+                }
+            }
+            DataSourceMetadataInfo metadataInfo = null;
+            DataSourceManager dataSourceManager = new DataSourceManager();
+            DataSourceInfo datasource = CommonUtils.getDataSourceInfo("prometheus-1");
```

Comment:
If you are making any assumptions on the datasource, move it all to a constants file and add a TODO section to use actual values in the input JSON and not make any assumptions

Reply:
done